Automattic/newspack-popups
# Display popups in the Customizer

### All Submissions:






### Changes proposed in this Pull Request:



Closes #577, alternative approach than #586.

As pointed out [in #586](https://github.com/Automattic/newspack-popups/pull/586#pullrequestreview-714657397), there's already a mechanism in place for preview session - and opening the Customizer is just that. This PR uses that approach, which is the same as in the Campaigns Wizard.

The biggest change is that _all_ active popups' data will be included in the request to the API (the `amp-access` endpoint). This is because with all the possible ways of inserting prompts, retrieving just the rendered ones gets unmanageable. The cost of a bigger payload is worth the gain in reduced complexity and reduced chance of missing a prompt in `amp-access` IMO.

One small, unrelated addition is the improvement of the debug message in 2f2abda98a71bacc78933c9975528b77038f4c20, which should be quite self-explanatory.

### How to test the changes in this Pull Request:

Similar to steps as in #586:

1. Publish three inline prompts. Make sure two are assigned to a custom placement.
2. Go to **Appearance > Customize > Widgets**. In any widget area, add a Single Prompt block and a Custom Placement block and select a prompt and custom placement containing prompts, respectively. In the Customizer preview, navigate to a page that displays the widget area with your blocks and observe that nothing is shown in the preview widget area.
3. Check out this branch, refresh the Customizer, and repeat step 2. This time confirm that one prompt is shown for each block, ignoring segmentation rules for previewing purposes.
4. Observe that the Customizer interaction is a separate "preview session" - if an overlay prompt is dismissed, that action is remembered, but the overlay will show up again on a subsequent Customizer session.

### Other information:
